### ISSUE TITLE
`keyedComposables` doesn't inject the key when a composable is imported in a module

### ISSUE THREAD

**Issue by CernyMatej** [nuxt/nuxt#26275]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.9.0`
- Nuxt Version:     `^3.10.3`
- CLI Version:      `3.10.1`
- Nitro Version:    `-`
- Package Manager:  `pnpm@8.15.1`
- Builder:          `-`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/nuxt-starter-k1ahtb?file=nuxt-project%2Fapp.vue,nuxt-module%2Fsrc%2Fruntime%2Fcomposables%2FuseGetKey.ts,nuxt-module%2Fsrc%2Fruntime%2Fcomposables%2FuseImportedGetKey.ts,nuxt-module%2Fsrc%2Fruntime%2Fcomposables%2FuseWrappedButNotImportedGetKey.ts

### Describe the bug

When a composable added to `optimization.keyedComposables` is explicitly imported within a module, the key is not passed as an argument to it.

**Steps to reproduce**:
- run the dev server in `nuxt-project/`
- the composable that uses an explicit import in the module doesn't have the auto-generated key

relevant files:
`nuxt-module/src/runtime/composables/useGetKey.ts`
`nuxt-module/src/runtime/composables/useImportedGetKey.ts`
`nuxt-module/src/runtime/composables/useWrappedButNotImportedGetKey.ts`

**Comment by danielroe** [comment]
This is because we explicitly check for imports so we don't inject keys into similarly named but different composables.

https://github.com/nuxt/nuxt/blob/860cfe16b3e2bda4e4e3e02a66d37921cbca406d/packages/vite/src/plugins/composable-keys.ts#L56-L76

**Comment by cernymatej** [comment]
I've revisited this issue and noticed that since the modules are installed after the `ComposableKeysPlugin`, it doesn't appear possible to add new `keyedComposables` from modules.

Could we address this by moving the code block at:
https://github.com/nuxt/nuxt/blob/784a28df97595ecf418d6fcdc4ff67f89d8e2d1c/packages/nuxt/src/core/nuxt.ts#L282-L289

to somewhere after this loop?
https://github.com/nuxt/nuxt/blob/784a28df97595ecf418d6fcdc4ff67f89d8e2d1c/packages/nuxt/src/core/nuxt.ts#L563-L569

Additionally, we could mention in the documentation that the keys are injected only for auto-imported functions and those imported from `#imports`.

**Comment by cjpearson** [comment]
As someone who isn't using auto-imports, having a way to work around this restriction would be a big help. :)

To avoid inadvertently affecting the wrong composable, could a `specifier` option help? 

For example, with the config

```ts
{
    "name": "useFoo",
    "argumentLength": 1,
    "specifier": "my-package",
},
```

this would match

```ts
import { useFoo } from 'my-package'
useFoo()
```

but this would not

```ts
import { useFoo } from 'other-package'
useFoo()
```

**Comment by cernymatej** [comment]
I was thinking about adding something similar to the config object. 👍

We should also consider the scenario where a module adds an entry to `keyedComposables` and the user wants to disable it. It shouldn't break the key injection within the module, though.

**Comment by cjpearson** [comment]
> As someone who isn't using auto-imports, having a way to work around this restriction would be a big help. :)
> 
> To avoid inadvertently affecting the wrong composable, could a `specifier` option help?
> 
> For example, with the config
> 
> {
>     "name": "useFoo",
>     "argumentLength": 1,
>     "specifier": "my-package",
> },
> 
> this would match
> 
> import { useFoo } from 'my-package'
> useFoo()
> 
> but this would not
> 
> import { useFoo } from 'other-package'
> useFoo()

Actually, taking another look at this, does this feature already exist as the `source` option? https://github.com/nuxt/nuxt/blob/317ced76c4577c870655984906320acdf9435cd0/packages/nuxt/src/core/plugins/composable-keys.ts#L16 

I can't find info in the documentation, but from the code it looks like it might work the same way.

**Comment by cernymatej** [comment]
Yes, it does. You can add a regex to it as well. Basically the import must either match the string you provide to `source` exactly, or the provided regex.

I've been looking into improving this and documenting it while working on https://github.com/nuxt/nuxt/issues/14736.

As it may be necessary to make some adjustments to how it currently works due to the mentioned issue, I would personally recommend holding off on documenting this for a while.

**Comment by cjpearson** [comment]
Thanks for the update! I'll hold off on that aspect then. But would it hurt if I make a PR to allow modules to update the `keyedComposables` in the meantime?

**Comment by cernymatej** [comment]
I think you can go ahead 👍🙂
